app/drivers/distributed_manager.go

```diff
@@ -534,6 +611,35 @@ func (d *DistributedManager) cleanupFreeInstances(ctx context.Context, pool *poo
 	return err
 }
 
+func (d *DistributedManager) cleanupFreeCapacity(ctx context.Context, pool *poolEntry, maxAgeFree time.Duration) error {
+	reservedCapacitiesForPool, err := d.capacityReservationStore.ListByPoolName(ctx, pool.Name)
+	if err != nil {
+		return fmt.Errorf("failed to list capacity reservations for pool=%q error: %w", pool.Name, err)
+	}
+
+	var capacitiesToDelete []*types.CapacityReservation
+
+	for _, capacityReservation := range reservedCapacitiesForPool {
+		inst, err := d.GetInstanceByStageID(ctx, pool.Name, capacityReservation.StageID)
+		if err != nil {
+			continue
+		}
+		if inst == nil || inst.ID == "" {
+			createdAt := time.Unix(inst.Started, 0)
```

Comment:
this can lead to nil pointer since you are checking if inst == nil || inst.ID == "".

Reply:
Yeah that was the initial intention, might have got confused while coding it, have changed it